### PR TITLE
OSOE-664: Downloading the Chrome WebDriver fails with Chrome >=15 in Lombiq.UITestingToolbox

### LINKED ISSUE #302: Downloading the Chrome WebDriver fails with Chrome >=15 (OSOE-664)
This is because WebDriverManager.Net didn't follow up with the URL of ChromeDriver packages having a different URL starting with v15: https://github.com/rosolko/WebDriverManager.Net/issues/253 The linked issue also has potential solutions. The task here is to fix our issue, in the following ways:

- [ ] Add a temporary fix first that we can get in to `dev` and out via NuGet ASAP, most possibly [this one](https://github.com/rosolko/WebDriverManager.Net/issues/253#issuecomment-1645953854) ([here's](https://github.com/RauCu/YoutubeDownloaderZ/commit/3d7b450c3f8eb5b867678d437b62a569c7bbd87c) some code that shows most possibly this in a wider context; we need to use it [here](https://github.com/Lombiq/UI-Testing-Toolbox/blob/f77f2d40f98b8ddc58081edeaea00168f9249dc1/Lombiq.Tests.UI/Services/WebDriverFactory.cs#L61)). Publish a new NuGet release.
- [ ] Implement a long-term fix:
  - Evaluate using [Selenium Manager](https://www.selenium.dev/documentation/selenium_manager/) instead of WebDriverManager.Net: that had this issue too (https://github.com/SeleniumHQ/selenium/pull/12208), which might not be released yet.
  - Evaluate using [Atata.WebDriverSetup](https://github.com/atata-framework/atata-webdriversetup).
  - If WebDriverManager.Net is still the best choice, then fork it and submit a fix.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-664)

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OSOE-664](https://lombiq.atlassian.net/browse/OSOE-664)
Fixes #302

[OSOE-664]: https://lombiq.atlassian.net/browse/OSOE-664?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.Tests.UI/Services/WebDriverFactory.cs [hunk 1]
```diff
@@ -173,20 +171,10 @@ private static TDriver CreateDriver<TDriver>(IDriverConfig driverConfig, Func<TD
...
-
             // While SetUpDriver() does locking and caches the driver it's faster not to do any of that if the setup was
             // already done. For 100 such calls it's around 16s vs <100ms. The Lazy<T> trick taken from:
             // https://stackoverflow.com/a/31637510/220230
-            _ = _driverSetups.GetOrAdd(driverConfig.GetName(), _ => new Lazy<bool>(() =>
-            {
-                new DriverManager().SetUpDriver(driverConfig, version);
-                return true;
-            })).Value;
+            version = _driverSetups.GetOrAdd(browserName, name => new Lazy<string>(() => DriverSetup.AutoSetUp(name).BrowserName)).Value;
```
Comment (Piedone) [inline Lombiq.Tests.UI/Services/WebDriverFactory.cs]: Is this kind of caching still needed?

File: Lombiq.Tests.UI/Services/WebDriverFactory.cs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+using Atata.WebDriverSetup;
```
Comment (Piedone) [inline Lombiq.Tests.UI/Services/WebDriverFactory.cs:1]: Also update _Tools.md_.

**Review by Piedone — CHANGES_REQUESTED** [review]
Ah, you already jumped ahead, nice!

File: Lombiq.Tests.UI/Services/WebDriverFactory.cs (hunk 1 above)
Comment (sarahelsaig) [reply Lombiq.Tests.UI/Services/WebDriverFactory.cs]: [I did some measurements](https://github.com/Lombiq/UI-Testing-Toolbox/pull/303/commits/6e96e930da403d5ddb3369227f2a62b11a0da33b) and I don't think it's needed any more.

Comment (Piedone) [reply Lombiq.Tests.UI/Services/WebDriverFactory.cs]: Great then!

**Review by Piedone — APPROVED** [review]

**Pull request merged**